common/sgx/collateral.h

```diff
@@ -45,6 +45,8 @@ oe_result_t oe_validate_revocation_list(
  *
  * @param[in] leaf_cert The PCK certificate.
  * @param[out] args Quote verification collateral.
+ * @deprecated Use
+ * oe_get_sgx_quote_verification_collateral_with_params_from_certs instead
```

Comment:
Isn't this file a purely internal header, not published in the SDK?   https://github.com/openenclave/openenclave/blob/master/docs/ApiGuidelines.md explains:

> Experimental and internal APIs can have breaking changes across releases

Thus you should not need to deprecate these, you can just delete them.

Reply:
Cool, just realized this file is not public, is there any doc about how can identify whether a header file is public or not?

Reply:
The usual rule is:
* headers not under openenclave/include are internal
* headers under openenclave/include/openenclave/internal are also internal
* other headers under openenclave/include are public

but openenclave/include/openenclave/bits/ has a mix it seems, which is issue #4132

And of course issue #4370 is also relevant if there are internal APIs in any public headers.

Reply:
Thanks, so in this sense, all edl files are also treated as public APIs

Reply:
Filed #4388